target/portfolio-1/js/playGame.js

```diff
@@ -7,34 +7,106 @@ async function initMapPlayGame() {
+    var panoramaOptions = {
+      enableCloseButton:false
+    };
+    panorama.setOptions(panoramaOptions);
     panorama.setPosition(startingLocation);
-    panorama.setPov(/** @type {google.maps.StreetViewPov} */({
-      heading: 265,
-      pitch: 0
-    }));
     panorama.setVisible(true);
 
+    createGameInfoOnSideOfMap(data, stage1);
+  });
+}
+
+function createGameInfoOnSideOfMap(data, stage) {
     var gameInfo = document.getElementById('game-info');
     
     var gameName = document.createElement('h1');
     gameName.innerHTML = data.gameName;
     gameName.className = 'center'
     gameInfo.appendChild(gameName);
-    
-    
-  });
+
+    var gameStage = document.createElement('h2');
+    gameStage.innerHTML = 'You are on stage:' + stage.stageNumber + '/' + data.stages.length;
+    gameStage.className = 'center'
+    gameInfo.appendChild(gameStage);
+
+    var theWordHints = document.createElement('h3');
+    theWordHints.innerHTML = 'Hints:';
+    gameInfo.appendChild(theWordHints);
+
+    var starterHint = document.createElement('h3');
+    starterHint.innerHTML = 'Starter: ' + stage.startingHint;
+    gameInfo.appendChild(starterHint);
+
+    var hintsDiv = document.createElement('div');
+    var hintsOl = document.createElement('ol');
+    hintsOl.id = 'hints';
+
+    getHints(stage).forEach(hint => 
+      hintsOl.appendChild(createHintPlaceHolder(hint.hintNumber))
+    );
+
+    hintsDiv.appendChild(hintsOl);
+    gameInfo.appendChild(hintsDiv);
+
+    var enterKeyText = document.createElement('h2');
+    enterKeyText.innerHTML = 'Please Enter The Key To Complete The Stage:';
+    enterKeyText.className = 'center'
+    gameInfo.appendChild(enterKeyText);
+
+    var inputKeyBox = document.createElement('input');
+    inputKeyBox.type = 'text';
+    inputKeyBox.className = 'center';
+    inputKeyBox.classList = 'input-text-color';
+    inputKeyBox.id = 'key-input';
+    inputKeyBox.style = 'width: 30%';
+
+    // This checks if the user clicked enter in the key box
+    inputKeyBox.addEventListener('keydown', function(e) {
+      if (e.which == 13) {
+        checkKey(data, stage.key);
+      }
+    });
+    gameInfo.appendChild(inputKeyBox);
+
+    var buttonToCheckKey = document.createElement('input');
+    buttonToCheckKey.type = 'button';
+    buttonToCheckKey.className = 'center';
+    buttonToCheckKey.value = 'Submit';
+    buttonToCheckKey.addEventListener('click', function() {
+      checkKey(data, stage.key);
+    });
+    gameInfo.appendChild(buttonToCheckKey);
+}
+
+//TODO: Change this so if there is more than one page and the input key is correct,
+// it adds the new stage rather than redirecting to the after game page
+function checkKey(data, key) {
+  var keyInput = document.getElementById('key-input');
+  var inputValue = keyInput.value;
+
+  if (key == inputValue ) {
+      window.location.replace('afterGame.html');
+  } else {
+      window.alert('Wrong key, please try again!');
```

Comment:
Not an urgent issue but this might not be the best way to indicate that the key is wrong. It forces the user to move their hand to the mouse to click on the alert box, which is extra effort. It might be better to just put the message under the key input box.

Again, not a big deal and can be adjusted later.

Reply:
I agree, with that and I was thinking of doing that as well. I only have it that way since it is still the prototype and just to get the functionality down first. I will change this later on.